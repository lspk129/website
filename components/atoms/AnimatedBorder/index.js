import styled from 'styled-components';

const AnimatedBorder = styled.div`
  color: ${({ theme }) => theme.text};
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 16px;
  cursor: pointer;
  &:after {
    display: grid;
    content: '';
    border-bottom: solid 2px ${({ theme }) => theme.text};  
    transform: scaleX(0);  
    transition: transform 250ms ease-in-out;
  }
  &:hover:after {
    transform: scaleX(1);
  }

`;

export default AnimatedBorder;
