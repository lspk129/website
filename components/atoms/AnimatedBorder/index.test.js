import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AnimatedBorder from './';

describe('AnimatedBorder', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<AnimatedBorder />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders with text color props', () => {
    const wrapper = renderer.create(<AnimatedBorder color={'white'} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders div component', () => {
    const wrapper = shallow(<AnimatedBorder />);
    expect(wrapper).toBePresent();
    expect(wrapper).toHaveTagName('div');
  });
});
