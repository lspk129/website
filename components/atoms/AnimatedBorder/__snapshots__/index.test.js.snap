// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedBorder renders correctly 1`] = `
<div
  className="sc-bdVaJa udBkl"
/>
`;

exports[`AnimatedBorder renders with text color props 1`] = `
<div
  className="sc-bdVaJa kUbMBk"
  color="white"
/>
`;
